src/main.cpp

```diff
@@ -327,6 +327,38 @@ void setMotorSpeeds(Speeds requestedMotorSpeeds, Servo &motorLeft, Servo &motorR
     motorRight.writeMicroseconds(map(commandMotorSpeeds.right * -1, -100, 100, 1000, 2000));
 }
 
+void nudge(bool leftForward = true, bool rightForward = true, float minChange = 5) {
+    Speeds motorPower, direction;
+    float nudgePower = 20;       //determined from practical testing
+    
+    direction.left = leftForward ? 1 : -1;
+    direction.right = rightForward ? 1 : -1;
+    
+
+    //assign a small value that should be enough to move, but not quickly
+    motorPower.left = nudgePower * direction.left;
+    motorPower.right = nudgePower * direction.right;
+    motorLeft.writeMicroseconds(map(motorPower.left, -100, 100, 1000, 2000));
+    motorRight.writeMicroseconds(map(motorPower.right * -1, -100, 100, 1000, 2000));
+
+    float change;
+    long encoderChange[NUM_ENCODERS];
+    while (change<minChange){
+        /// Read Encoder counts
+        for (u_int8_t n = 0; n < 3; n++) {
+            encoderChange[n] = encoders[n].read()-encoderReadings[n];
+        }
+        change = direction.left * (encoderChange[0]+encoderChange[1]+encoderChange[2]);
+        change -= direction.right * (encoderChange[3]+encoderChange[4]+encoderChange[5]);
+        delay(1);
```

Comment:
Hmm. I see what you mean about the `delay()` - 1ms is a chunk of our loop time budget. I'll stick this one on the back burner and give it some thought.

Reply:
the delay isn't required as such, it just needs to not take 30ms before it comes back to this. To be honest, I don't know how long it can be and nudge still work ok. I figured with this approach it would stop as soon as possible after it had moved the desired amount. which for the front/back nudges is 3encoder counts so may be very quick. the turn nudge only worked with an encoder count of ~100, which I assume means it takes quite a while (relatively speaking). 
I can try to make a non-blocking variant, but I think it will need a 'state' variable creating. which would be useful for other things.